### ISSUE TITLE
SuperSource reports all boxes at default values

### ISSUE THREAD

**Issue by kellyzdude** [clvLabs/PyATEMMax#9]
I'm utilizing PyATEMMax in an effort to set SuperSource settings, but I'm also trying to gather the current SuperSource state before making changes.

I have this block of code pulling SuperSource state from my ATEM Mini Extreme ISO:
```
        for box in range(zboxes):
                zvalues[box] = {}
                zvalues[box]['enabled'] = switcher.superSource.boxParameters[box].enabled
                zvalues[box]['source'] = switcher.superSource.boxParameters[box].inputSource
                zvalues[box]['posx'] = switcher.superSource.boxParameters[box].position.x
                zvalues[box]['posy'] = switcher.superSource.boxParameters[box].position.y
                zvalues[box]['size'] = switcher.superSource.boxParameters[box].size
                zvalues[box]['crop'] = switcher.superSource.boxParameters[box].cropped
                zvalues[box]['croptop'] = switcher.superSource.boxParameters[box].crop.top
                zvalues[box]['cropbottom'] = switcher.superSource.boxParameters[box].crop.bottom
                zvalues[box]['cropleft'] = switcher.superSource.boxParameters[box].crop.left
                zvalues[box]['cropright'] = switcher.superSource.boxParameters[box].crop.right
        pprint.pprint(zvalues)
```

It gets me this output:
```
{0: {'crop': False,
     'cropbottom': 0.0,
     'cropleft': 0.0,
     'cropright': 0.0,
     'croptop': 0.0,
     'enabled': False,
     'posx': 0.0,
     'posy': 0.0,
     'size': 0.0,
     'source': PyATEMMax.ATEMProtocolEnums.ATEMConstant(name=, value=None)},
 1: {'crop': False,
     'cropbottom': 0.0,
     'cropleft': 0.0,
     'cropright': 0.0,
     'croptop': 0.0,
     'enabled': False,
     'posx': 0.0,
     'posy': 0.0,
     'size': 0.0,
     'source': PyATEMMax.ATEMProtocolEnums.ATEMConstant(name=, value=None)},
 2: {'crop': False,
     'cropbottom': 0.0,
     'cropleft': 0.0,
     'cropright': 0.0,
     'croptop': 0.0,
     'enabled': False,
     'posx': 0.0,
     'posy': 0.0,
     'size': 0.0,
     'source': PyATEMMax.ATEMProtocolEnums.ATEMConstant(name=, value=None)},
 3: {'crop': False,
     'cropbottom': 0.0,
     'cropleft': 0.0,
     'cropright': 0.0,
     'croptop': 0.0,
     'enabled': False,
     'posx': 0.0,
     'posy': 0.0,
     'size': 0.0,
     'source': PyATEMMax.ATEMProtocolEnums.ATEMConstant(name=, value=None)}}
```

Which 1) does not match the ATEM configuration (boxes 1 and 2 are enabled, both are scaled, and have non-0,0 positions), and 2) looks an awful lot like default values are being output.

Is this a defect in the module, or am I doing something wrong in gathering these details?

```
$ pip3 show pyatemmax
Name: PyATEMMax
Version: 1.0b5
Summary: A Python library to monitor and control Blackmagic Design ATEM video switchers.
Home-page: https://github.com/clvLabs/PyATEMMax
Author: Tony Aguilar
Author-email: onixag@gmail.com
License: GPL
Location: /home/administrator/.local/lib/python3.6/site-packages
Requires:
$
```

```
$ python3 --version
Python 3.6.9
```

I have not yet tried setting SuperSource parameters -- maintaining a cache of what I've told SuperSource to do would be a viable, though not-preferred workaround.

Happy to share any additional debug data if needed, just let me know what I need to provide.

**Comment by kellyzdude** [comment]
No problem! Thanks for the feedback.

[logLevelDebug.txt](https://github.com/clvLabs/PyATEMMax/files/7377853/logLevelDebug.txt)

I see the _top, I also see _SSC a little further down
02:59:19.907 DEBUG    [ATEMConnectionManager] Received: [_top] (Topology)
02:59:19.908 DEBUG    [ATEMConnectionManager] Received: [_SSC] (Super Source Config)

I see multiple InPr messages:
02:59:19.912 DEBUG    [ATEMConnectionManager] Received: [InPr] (Input Properties)

I also see multiple SSBP/SSRC messages, though they indicate errors in the data:
02:59:19.994 DEBUG    [ATEMConnectionManager] Received: [SSrc] (Super Source)
02:59:19.994 DEBUG    [ATEMConnectionManager] Received: [SSBP] (Super Source Box Parameters)
02:59:19.995 WARNING  [ATEMConnectionManager] UNKNOWN ATEMBoxes 30 - processing [SSBP]
02:59:19.995 DEBUG    [ATEMConnectionManager] Received: [SSBP] (Super Source Box Parameters)
02:59:19.995 WARNING  [ATEMConnectionManager] UNKNOWN ATEMBoxes 244 - processing [SSBP]
02:59:19.995 DEBUG    [ATEMConnectionManager] Received: [SSBP] (Super Source Box Parameters)
02:59:19.996 WARNING  [ATEMConnectionManager] UNKNOWN ATEMBoxes 101 - processing [SSBP]
02:59:19.996 DEBUG    [ATEMConnectionManager] Received: [SSBP] (Super Source Box Parameters)
02:59:19.996 WARNING  [ATEMConnectionManager] UNKNOWN ATEMBoxes 110 - processing [SSBP]

Looking at the Socket level:
[logCombined.txt](https://github.com/clvLabs/PyATEMMax/files/7377887/logCombined.txt)

To be honest, I'm not sure what I'm looking at. It does appear to have the extra socket level debug data, but I don't know how to read it.

As far as protocol version -- it's possible. 

To be clear, the four boxes I'm looking at are representing boxes on the SuperSource layout -- the Extreme only has a single SuperSource available, but it presents up to four video sources on the view. I'd like to be able to move those around on the canvas, but ideally I need to first know what the current state is.

**Comment by clvLabs** [comment]
I think I found it!

Please bear with me... I'll release a new version with this bug fixed as soon as I can.

If you're in a hurry and feel brave about making changes in the library's code, it's just 1 byte (I guess/hope):
In `ATEMCommandHandlers.py`, [line 755](https://github.com/clvLabs/PyATEMMax/blob/7b2fc460d90fcc41118ca9f73a3781000f937033/PyATEMMax/ATEMCommandHandlers.py#L755) you'll find:
```python
        box = self._getBufEnum(3, 8, self._p.boxes)
```
Change the first parameter from `3` to `0`:
```python
        box = self._getBufEnum(0, 8, self._p.boxes)
```
... and you should be ready to go!

If you make this change, please let me know. Otherwise you'll have a new version soon, as I said.

**Comment by clvLabs** [comment]
Besides that... while reading the `debug` dump you sent me I found something strange.

According to the [original protocol description](https://www.skaarhoj.com/discover/blackmagic-atem-switcher-protocol), inside the `SSBP` message :
* Byte 10 contains the `Cropped` field
* Byte 11 is tagged as `unknown`

The `SSBP` messages in your debug log contain:
|Box|Enabled|Input Source|Position X|Position Y|Size|Cropped|(Unknown)|Crop Top|Crop Bottom|Crop Left|Crop Right|
|---|---|---|---|---|---|---|---|---|---|---|---|
|`00` (box1)|`01` (ON)|`1e00` (30)|`03fb` (64259)|`1e01` (286)|`5e01` (350)|`f4` (244)|`01` (ON)|`0100` (1)|`0000` (0)|`0027` (9984)|`1027` (10000)|
|`01` (box2)|`01` (ON)|`f400` (244)|`0201` (258)|`2c00` (44)|`0002` (512)|`ee` (238)|`00` (OFF)|`c200` (194)|`0000` (0)|`0000` (0)|`0000` (0)|
|`02` (box3)|`01` (ON)|`650b` (2917)|`c3fb` (64451)|`16fe` (65046)|`5701` (343)|`2c` (44)|`00` (OFF)|`c000` (192)|`0000` (0)|`0000` (0)|`0000` (0)|
|`03` (box4)|`00` (OFF)|`6e00` (6)|`0000` (0)|`0000` (0)|`0003` (768)|`e8` (232)|`00` (OFF)|`6f00` (111)|`0000` (0)|`0000` (0)|`0000` (0)|

This makes me think that:
* The documentation has bytes `10` and `11` reversed.
* Your `Box 1` was indeed cropped while the others were not.
* There is really some information in that `unknown` field (sad we don't know what it is)

Is the information you see in the table right in any way ?

**Comment by clvLabs** [comment]
Version `1.0b6` is already available on [pypi](https://pypi.org/project/PyATEMMax/), you can do a `pip install PyATEMMax --upgrade`

I won't close the issue yet, still waiting for your response on my last comment :)

**Comment by clvLabs** [comment]
Just got a [new pull request](https://github.com/clvLabs/PyATEMMax/pull/13) that seems to be related with this, looks like good news! 😄

**Comment by clvLabs** [comment]
Hi there, @kellyzdude !
I just merged the pull request I was mentioning and just published [version 1.0b7](https://pypi.org/project/PyATEMMax/) on pypi.org.
I am almost sure this will solve your issues. 
Please let me know so I can close this issue  :)
